bin/desi_tsnr_afterburner

```diff
@@ -210,99 +343,44 @@ def main():
-                table.meta['TILEID'] = tileid
-                table.meta['CAMERA'] = camera
-                table.meta['EXTNAME'] = 'TSNR2'
-
-                #- Write per-expid per-camera output if requested
-                if compute_tsnr and table_output_filename is not None :
-                    Path(os.path.dirname(table_output_filename)).mkdir(parents=True,exist_ok=True)
-                    tmpfile = table_output_filename +'.tmp'
-                    table.write(tmpfile, format='fits', overwrite=True)
-                    os.rename(tmpfile, table_output_filename)
-                    log.debug('Successfully wrote {}.'.format(table_output_filename))
-
-                #- Append to summary.
-                entry = dict()
-                entry['NIGHT'] = np.int32(night)
-                entry['EXPID'] = np.int32(expid)
-                entry['TILEID'] = np.int32(tileid)
-                entry['CAMERA'] = camera
-                for key in table.colnames:
-                    if key.startswith('TSNR2_'):
-                        #- TSNR2_{TRACER}_{BAND} -> TSNR2_{TRACER}
-                        short_key = '_'.join(key.split('_')[0:2])
-                        entry[short_key]=np.median(table[key]).astype(np.float32)
-                    if key.startswith('TSNR2_ALPHA'):
-                        entry['TSNR2_ALPHA'] = np.median(table[key]).astype(np.float32)
-
-                summary_rows.append(entry)
-                night_summary_rows.append(entry)
-                cframe_hdulist.close()
-
-        # end of loop on exposures for a night
-
-        # write one summary per night
-        #output_filename="{}-night-{}.fits".format(args.outfile.replace(".fits",""),night)
-        #write_summary(night_summary_rows,output_filename)
+                func_args.append({'prod':args.prod,'night':night,'expid':expid,'camera':camera,
+                                  'recompute':args.recompute,'alpha_only':args.alpha_only,'details_dir':args.details_dir})
+
+        if args.nproc == 1 :
+            for func_arg in func_args :
+                entry = func(**func_arg)
+                if entry is not None :
+                    summary_rows.append(entry)
+        else :
+            log.info("Multiprocessing with {} procs".format(args.nproc))
+            pool = multiprocessing.Pool(args.nproc)
+            results  =  pool.map(_func, func_args)
+            for entry in results :
+                if entry is not None :
+                    summary_rows.append(entry)
+            pool.close()
+            pool.join()
+
+
+        # write result after every other night
+        if len(summary_rows)>0 :
+
+            print("summary_rows = ",summary_rows)
+
+            tmpfilename=args.outfile.replace(".fits","_tmp.fits")
+            write_summary(summary_rows,tmpfilename,efftime_config,preexisting_tsnr2_expid_table,preexisting_tsnr2_frame_table)
+            log.info("wrote {} entries in tmp file {}".format(len(summary_rows),tmpfilename))
```

Comment:
This `tmpfilename` is written, but I don't see it ever renamed to a non-temp filename, or otherwise removed after the final file is written.  Am I missing something?